### PR TITLE
Add target libraries/executables to executable's RPATH

### PR DESCRIPTION
**Pull request by billysuh7**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

These files will be patched:
targets/x86_64-linux/bin/__nvcc_device_query
targets/x86_64-linux/bin/bin2c
targets/x86_64-linux/bin/cudafe++
targets/x86_64-linux/bin/fatbinary
targets/x86_64-linux/bin/nvcc
targets/x86_64-linux/bin/nvlink
targets/x86_64-linux/bin/ptxas

These will be skipped:
patchelf
targets/x86_64-linux/bin/crt/link.stub
targets/x86_64-linux/bin/crt/prelink.stub
targets/x86_64-linux/bin/nvcc.profile

Added **recipe/patchelf_exclude.txt** to exclude files to patch as they aren't binaries. The conda build will fail if a new binary is added without updating this file.

xref: https://github.com/conda-forge/cuda-feedstock/issues/10

### CODE REVIEW

**Comment by jakirkham** [comment]
Seeing the following warning [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1068176&view=logs&j=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&t=9eb77fd2-8ddd-5444-8fc0-71cb28dcb736&l=1229 )

```
WARNING (cuda-nvvm-tools,nvvm/bin/cicc): $RPATH/libstdc++.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
```

Can we take a closer look at how `cicc` is being patched and make sure it is doing the right thing?

File: recipe/build.sh [hunk 1]
```diff
@@ -29,6 +41,18 @@ for i in `ls`; do
         elif [[ $i == "include" ]]; then
             cp -rv $i ${PREFIX}/${targetsDir}
         elif [[ $i == "nvvm" ]]; then
+            for j in `find "${i}"`; do
+                if [[ "${j}" =~ /bin/.*$ ]]; then
+                    echo patchelf --force-rpath --set-rpath "\$ORIGIN/../lib64:\$ORIGIN/../../lib:\$ORIGIN/../../${targetsDir}/lib" "${j}" ...
```
Comment (jakirkham) [inline recipe/build.sh:46]: Suggesting a comment here to make it easier to understand what this is checking

```suggestion
                    # Adds the following paths relative to `$PREFIX` to the `RPATH`: `nvvm/lib64`, `lib`, `${targetsDir}/lib`
                    echo patchelf --force-rpath --set-rpath "\$ORIGIN/../lib64:\$ORIGIN/../../lib:\$ORIGIN/../../${targetsDir}/lib" "${j}" ...
```

File: recipe/build.sh [hunk 2]
```diff
@@ -29,6 +41,18 @@ for i in `ls`; do
...
+                    patchelf --force-rpath --set-rpath "\$ORIGIN/../lib64:\$ORIGIN/../../lib:\$ORIGIN/../../${targetsDir}/lib" "${j}"
+
+                # Do not patch libnvvm.so for now as it gets corrupted by patchelf. See build.sh for details
+                #elif [[ "${j}" =~ /lib.*/.*\.so($|\.) && ! -L "${j}" ]]; then
+                #    echo patchelf --force-rpath --set-rpath "\$ORIGIN" "${j}" ...
+                #    patchelf --force-rpath --set-rpath "\$ORIGIN" "${j}"
```
Comment (jakirkham) [inline recipe/build.sh:52]: This comment references `build.sh` for details however this is in `build.sh`. Are there more details we should add here or can that portion be dropped?

**Review by jakirkham — COMMENTED** [review]
Thanks Billy! 🙏

Had a couple minor suggestions below

Generally this seems reasonable

**Comment by billysuh7** [comment]
> Seeing the following warning [on CI](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1068176&view=logs&j=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&t=9eb77fd2-8ddd-5444-8fc0-71cb28dcb736&l=1229)
> 
> ```
> WARNING (cuda-nvvm-tools,nvvm/bin/cicc): $RPATH/libstdc++.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist.
> .. is this binary repackaging?
> ```
> 
> Can we take a closer look at how `cicc` is being patched and make sure it is doing the right thing?

cicc was not being patched - it is all set now

File: recipe/build.sh (hunk 2 above)
Comment (billysuh7) [reply recipe/build.sh]: Done. Ready for review.

**Comment by billysuh7** [comment]
The outstanding questions have been resolved.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Billy! 🙏

**Pull request merged**